mailbox/api/src/main/java/org/apache/james/mailbox/model/MailboxPath.java

```diff
@@ -242,4 +244,7 @@ public String getFullName(char delimiter) {
         return namespace + delimiter + name;
     }
 
+    public boolean hasParent() {
+        return getHierarchyLevels(DEFAULT_DELIMITER).size() > 1;
```

Comment:
1. You should retrieve the delimiter from the session, not use a static one. I suggest having a `char delimiter` parameter just like in getHierarchyLevels

2. No tests?